private_histograms_mpc.md

```diff
@@ -0,0 +1,368 @@
+*   Values are hidden from the helper servers via secret shares
+*   Keys are hidden from the helper servers via the homomorphic encryption
+*   Helpers learn noisy counts of keys in pseudorandom space AggID = H(aggkey)<sup>s1*s2</sup>. We can achieve DP on counts of keys in the output domain (see [analysis](#intermediate-dp-analysis)).
+    *   Counts for keys not in the output are _not_ protected by DP, but they are also never revealed in the clear, only in pseudorandom space.
+*   If one helper is malicious and colluding with an ad-tech
+    *   DP with partial noise for sums of values and credit (helper reveals their noise share added to the sums). 
+    *   Internal counts of records associated with each AggID is learned, artificially inflated by the fake records (can be analyzed with some  [flavor of DP](#intermediate-dp-analysis))
+    *   Records that don’t come from the advertised output domain are learned in pseudorandom domain but the aggkey cannot be recovered
+
+
+## Intermediate DP analysis
+
+We can achieve DP bounds on the internal counts of records associated with a given AggID, as long as it is in the output domain. We do this by sampling fake records in the output domain and injecting them into the protocol. That is, we have a mechanism to _add_ to a sensitive count, but not subtract.
+
+How to configure _p_ to achieve DP depends on the kind of differential privacy bounds we want to show. We have a few potential options.
+
+### Epsilon / delta DP
+
+The problem here is similar to the problem of [differential privacy with constraints](https://arxiv.org/pdf/2007.01181.pdf). In our case, the constraint is that the noisy count must always be greater than the true count. To achieve (eps,delta)-DP in this scenario we need to add Z fake records where: 
+
+```
+Z = s + L
+s = (1/eps) log(((exp(eps) - 1) / (2 delta)) + 1)
+L = Laplace(1/eps_sum) truncated to (-s, s)
+```
+
+After applying the noise, the result must be rounded to the nearest whole number in a post-processing step.
+
+This will yield s fake records per AggKey in expectation. For example, choosing eps = ln(2) and delta = 10^-6 will yield s = ~19 expected fake records per key.
+
+
+### One-sided DP
+
+See [this paper](https://arxiv.org/pdf/1712.05888.pdf) for more information on one-sided DP. In this formulation we consider the presence of a conversion to be sensitive, and the non-presence of a conversion to be not sensitive. This may be a reasonable relaxation of DP in the event of one malicious helper.
+
+The basic tool here will be adding fake records distributed according to a Geometric distribution.
+
+```
+P(Geo(p) == k) = p(1-p)^k
+```
+
+
+Consider an AggID count R and R’ that are one-sided-dp-neighbors. wlog there is only one such neighbor to consider, where R has count = n + 1 and R’ has count = n. That is, R’ is a one-sided DP neighbor of R because it consists of replacing a sensitive record with its absence.
```

Comment:
be consistent with line below. Is wlog meant to be there?

Reply:
wlog = without loss of generality. I'll spell it out.

Reply:
I think the correct hyphen rules would say "one-sided-DP neighbors" and "one-sided DP"